### PR TITLE
Adding extended tests for Ruby and PHP hot deploy

### PR DESCRIPTION
**Pull request by jhadvig**
@mnagy @mfojtik PTAL

### CODE REVIEW

**Comment by mnagy** [comment]
[test][extended:core(images: s2i)]

File: test/extended/images/helper.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ func GetImageLabels(c client.ImageStreamImageInterface, imageRepoName, imageRef
 }
 
 // ModifySourceCode will modify source code in the pod of the application
-// according to the sed script.
-func ModifySourceCode(oc *exutil.CLI, selector labels.Selector, sedScript, file string) error {
+// according to the provided command.
+func ModifySourceCode(oc *exutil.CLI, selector labels.Selector, cmd []string) error {
```
Comment (mfojtik) [inline test/extended/images/helper.go]: why it is an array and not just a string?

Comment (mnagy) [inline test/extended/images/helper.go]: I think array is more consistent with rest of our code (and code everywhere else in the universe), plus it frees you from too much quoting when everything is in a string.

Comment (jhadvig) [inline test/extended/images/helper.go]: I had the string at the beginning but then I've realized that array is more readable and as @mnagy mentioned also more consistent with rest of the code

Comment (mnagy) [inline test/extended/images/helper.go]: But I do not understand this change in general. You don't end up using a different command than `sed -ie <script> <file>`. The `ModifySourceCode` suddenly doesn't seem as an appropriate name.

Comment (jhadvig) [inline test/extended/images/helper.go]: In the ruby extended test I actually am removing `public/index.html` file with that function. Why it's not appropriate ? Deleting file is a modification as well.

Comment (jhadvig) [inline test/extended/images/helper.go]: But if it makes you happier I can change it to something like `ExecuteCommand` or `ExecuteContainerCommand` :-)

**Comment by deads2k** [comment]
jenkins issue re[test]

**Comment by jhadvig** [comment]
re[test]

**Comment by mnagy** [comment]
The Jenkins is passing the hot deploy test, the one failure is for MySQL replication tests. Looks like a flake, I will look into it.

@mfojtik it seems the tests are testing images, not s2i images. The build parameter is "`core(images:`" and not "`core(images: s2i)`" as I specified in my comment above. Is this my fault for not putting it in quotes, or is this a bug?

**Comment by jhadvig** [comment]
@mnagy I've run the MySQL tests locally and they passed without any issue.

**Comment by jhadvig** [comment]
re[test]

File: test/extended/images/s2i_ruby.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("images: s2i: ruby", func() {
+	defer g.GinkgoRecover()
+	var (
+		railsTemplate = "https://raw.githubusercontent.com/openshift/rails-ex/master/openshift/templates/rails-postgresql.json"
+		oc            = exutil.NewCLI("s2i-php", exutil.KubeConfigPath())
+		modifyCommand = []string{"sed", "-ie", `s%render :file => 'public/index.html'%%`, "app/controllers/welcome_controller.rb"}
+		removeCommand = []string{"rm", "-f", "public/index.html"}
```
Comment (mnagy) [inline test/extended/images/s2i_ruby.go]: Why are you removing the html file?

**Comment by mnagy** [comment]
@danmcp please see my comment: https://github.com/openshift/origin/pull/4988#issuecomment-146464576

File: test/extended/images/s2i_ruby.go (hunk 2 above)
Comment (jhadvig) [inline test/extended/images/s2i_ruby.go]: Cause in rails the public/index.html is served by default when requiring the app root. You have to delete it so other views can get served.

**Comment by mnagy** [comment]
Restarted the test with fixed build parameters: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/5550/

**Comment by mnagy** [comment]
One more manual rebuild, this time with quoted focus: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/5551/

**Comment by mnagy** [comment]
LGTM :ok_hand:

**Comment by mfojtik** [comment]
[merge]

**Comment by mnagy** [comment]
So.. Quotes are not enough..
```
The machine with the name 's2i)"' was not found configured for
```

**Comment by mfojtik** [comment]
@mnagy this is a bug in the test_pull_request parser, do not try to workaround it @danmcp FYI

**Comment by mnagy** [comment]
@mfojtik yeah, lesson learned.

**Comment by danmcp** [comment]
@mnagy @mfojtik Should be fixed.

**Pull request merged**